CMakeLists.txt

```diff
@@ -65,33 +93,37 @@ function(trusted_firmware_build)
     tfm
     SOURCE_DIR ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m
     BINARY_DIR ${TFM_BINARY_DIR}
-    CMAKE_ARGS -DPROJ_CONFIG=${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/configs/${TFM_CFGFILE}.cmake
-               -DTARGET_PLATFORM=${TFM_BOARD}
+    CMAKE_ARGS -DCMAKE_TOOLCHAIN_FILE=${ZEPHYR_TFM_MODULE_DIR}/${CMAKE_TOOLCHAIN_FILE}
+               -DTFM_PLATFORM=${TFM_BOARD}
+               ${TFM_CMAKE_BUILD_TYPE_ARG}
                ${TFM_BL2_ARG}
-               -DCOMPILER=${TFM_TOOLCHAIN}
-               -DGNUARM_PREFIX=${TFM_TOOLCHAIN_PREFIX}
-               -DGNUARM_PATH=${TFM_TOOLCHAIN_PATH}
+               ${TFM_IPC_ARG}
+               ${TFM_ISOLATION_LEVEL_ARG}
+               ${TFM_REGRESSION_ARG}
+               -DTFM_TEST_REPO_PATH=${ZEPHYR_TFM_MODULE_DIR}/tf-m-tests
```

Comment:
I would also add -DNS=OFF, I see no point in building the TFM NS image, we use Zephyr as NS image and Zephyr's cmake handles signing and image merge. Saves some build time.

Reply:
Thanks @agansari , I would like to keep this out of the up-merge PR; this is sort of a change compared to the current state. But the proposal seems reasonable, so I will include this in a list of improvements for the TF-M module (I will open a dedicated ticket for that copy all TFM stakeholders)